jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -193,4 +191,23 @@ static FileCollection getInputFiles(File extraDirectory, Project project) {
       return project.files(dependencyFileCollections);
     }
   }
+
+  /**
+   * Validates and converts a {@code String->String} file-path-to-file-permissions map to an
+   * equivalent {@code AbsoluteUnixPath->FilePermission} map.
+   *
+   * @param inputMap the map to convert
+   * @return the converted map
+   */
+  @VisibleForTesting
+  static Map<AbsoluteUnixPath, FilePermissions> convertPermissionsMap(
+      Map<String, String> inputMap) {
+    ImmutableMap.Builder<AbsoluteUnixPath, FilePermissions> permissionsMap = ImmutableMap.builder();
```

Comment:
nit: I think we can just return a `HashMap`. I see no need to prevent callers from modifying the returned map.

Reply:
Oh yea, and for cases where we were to return `ImmutableMap`, we should be making the return type be `ImmutableMap`.